extensions/npm/src/tasks.ts

```diff
@@ -149,6 +149,32 @@ export async function getPackageManager(extensionContext: ExtensionContext, fold
 	return packageManagerName;
 }
 
+/**
+ * Returns an array of args to launch the package manager, which may include additional commands such as nvm-exec.
+ * @returns an array of args to launch the package manager
+ */
+export async function getPackageManagerLaunchArgs(extensionContext: ExtensionContext, folder: Uri, showWarning: boolean = true): Promise<string[]> {
+	const packageManager = await getPackageManager(extensionContext, folder, showWarning);
+	const nvmExecPath = getNvmExecPath(folder);
+	if (nvmExecPath) {
+		return [nvmExecPath, packageManager];
+	} else {
+		return [packageManager];
+	}
+}
+
+/**
+ * @returns the path to nvm-exec if nvm is used, or `undefined`
+ */
+export function getNvmExecPath(folder: Uri | undefined): string | undefined {
+	const useNvm = workspace.getConfiguration('npm', folder).get<boolean>('useNvm');
+	if (useNvm) {
+		return `${process.env['NVM_DIR'] || '~/.nvm'}/nvm-exec`;
```

Comment:
Ideally this would also support nvm for Windows if that has similar way of doing this.

Reply:
@alexr00 doh! of course... I've looked into this and it seems that the answer is "it's complicated"

https://github.com/coreybutler/nvm-windows/issues/128 suggests that `.nvmrc` files aren't something that `nvm-windows` will support. I wonder if people are using standard `nvm` in WSL instead, I don't support you or anyone watching uses `nvm` on Windows? It's part of the VS Code codebase right?

Reply:
I use nvm on Windows, but I'm a shallow user of it so I don't have any great knowledge here. I don't think this PR needs to add support for this, I just thought it was worth mentioning in case there was some easy way to do it.

Reply:
I would love to test how this works on WSL with `nvm` as I suspect it might work. I will attempt to test that today to see if we can (a) make it work and (b) specify that in the README.